### PR TITLE
Improve ray/AABB benchmarks

### PR DESCRIPTION
**Pull request by tavianator**
- Fix ray/AABB benchmarks
- Optimize ray/AABB intersection tests.

### CODE REVIEW

File: src/ray.rs [hunk 1]
```diff
@@ -63,6 +63,16 @@ impl Intersection {
...
 
+/// Fast minimum that matches SSE NaN semantics.
+fn min(x: f32, y: f32) -> f32 {
+    if x < y { x } else { y }
+}
+
+/// Fast maximum that matches SSE NaN semantics.
+fn max(x: f32, y: f32) -> f32 {
+    if x > y { x } else { y }
+}
```
Comment (svenstaro) [inline src/ray.rs:74]: Perhaps force `#[inline]`?

Comment (svenstaro) [reply src/ray.rs:74]: It might also be worth retaining the original note about performance.

**Comment by svenstaro** [comment]
Very good stuff, thanks! I always wondered about the benchmark results but never quite had the will to figure it out. Also great optimization. Do you reckon the optimization is only due to the compiler generating SSE for it now?

**Comment by svenstaro** [comment]
Looks like you're going to have to appease the CI, though. :) After you do, I'm happy to merge as is and cut a release.

**Comment by tavianator** [comment]
> Do you reckon the optimization is only due to the compiler generating SSE for it now?

The generated code for `x.min(y)` on baseline x86-64 is something like

```asm
        cmpunordss      xmm2, xmm0
        movaps  xmm3, xmm2
        andps   xmm3, xmm1
        minss   xmm1, xmm0
        andnps  xmm2, xmm1
        orps    xmm2, xmm3
```

For `if x < y { x } else { y }` it's just

```asm
        minss   xmm0, xmm1
```

**Comment by svenstaro** [comment]
It's a bit silly that the built in is so inefficient but there's probably a good reason for it.

**Comment by tavianator** [comment]
> It's a bit silly that the built in is so inefficient but there's probably a good reason for it.

Well, [these semantics](https://github.com/svenstaro/bvh/pull/87/files#diff-a3a9b4c318707d2160a0d62c757a060d873e6da159d7ba1653f3972bb4e1a88dR73-R76) are pretty surprising, so I get going with the more consistent ones for the stdlib

**Comment by svenstaro** [comment]
@tavianator Could you rebase? I fixed the lints and the CI.

**Comment by svenstaro** [comment]
Super thanks!

**Pull request merged**